### ISSUE TITLE
Repeated route path in APM

### ISSUE THREAD

**Issue by vinnymac** [DataDog/dd-trace-js#1455]
**Describe the bug**

In my express TS application I have a route, such as `/products`. APM is reporting this route, and many others in a strange way.

APM Heading
<img width="334" alt="Screen Shot 2021-06-28 at 3 48 20 PM" src="https://user-images.githubusercontent.com/1832781/123695237-5027a900-d828-11eb-8888-a6f0567978f6.png">

Tags
<img width="297" alt="Screen Shot 2021-06-28 at 3 48 32 PM" src="https://user-images.githubusercontent.com/1832781/123695229-4dc54f00-d828-11eb-9847-56cd3e768c90.png">

It's saying the route is actually `/products/products`, but it is not. It's somehow repeating routes, and the http tag is supporting this. I am not sure what causes this repeated route path, but it ends up splitting up the data. Some of the data is reported as `/products` and others are reported by APM as `/products/products`. Am I doing something wrong with my configuration of `dd-trace-js` below?

**Configuration**

```ts
import tracer from 'dd-trace';

tracer.init({
  enabled: process.env.TRACER_ENABLED,
  service: process.env.SERVICE,
  analytics: true,
  logInjection: true,
  runtimeMetrics: true,
});

tracer.use('express', {
  headers: ['user-agent', 'x-forwarded-for', 'host', 'referer'],
  blacklist: ['/'],
  hooks: {
    request(span?: opentracing.Span, req?: IncomingMessage) {
      const request = (req as unknown) as Request | undefined;

      if (span && request) {
        span.addTags({
          body: request.body,
          params: request.params,
          headers: request.headers,
          ip: request.ip,
          hostname: request.hostname,
        });
      }
    },
  },
});

// later start the actual express application

bootstrap();
```

**Environment**

* **Operation system: node:12.19.1-alpine**
* **Node version: 12.19.1**
* **Tracer version: 0.34.0**
* **Agent version: latest**

**Comment by rochdev** [comment]
Configuration looks fine. I've seen this issue in the past but we were never able to get a reproduction. Is this something you can reproduce in isolation with a snippet of code you could share?

**Comment by vinnymac** [comment]
I can try to tweak the configuration and reproduce it locally with debug logs on. I believe I saw this start recently after adding the second half of the configuration where I have `tracer.use('express', ...);`. Although I can't say what caused it for sure.

This started for my team within the last few weeks, that's all I know for certain at this time.

**Comment by maximeloizeau** [comment]
We thought we had the same issue but we discovered that we were initializing the Datadog tracer multiple times in the same app when importing a module. Each time the tracer was initialized it recorded repeated paths, for example initialized 3 times caused routes to be repeated 3 times: `/test/test/test` instead of `/test`

In case it helps other people with the same issue. I'm not sure if it is the same reason for vinnymac.

**Comment by maximeloizeau** [comment]
We created a shared NPM module containing a few helper functions, including the Datadog tracer in a file like this:

```
import datadogTracer from "dd-trace";

const isDatadogConfigured = (env: NodeJS.ProcessEnv): Boolean => {
  const { DD_ENV, DD_SERVICE } = env;
  return !!(DD_ENV && DD_SERVICE);
};

if (isDatadogConfigured(process.env)) {
  datadogTracer.init({
    logInjection: true,
  });
}

export const tracer = datadogTracer;
```

and we were importing 
```
import { tracer } from "our-module";
tracer;
```

But we had another module importing this module, i'll try to explain:
```
main-project
|-- node_modules
     |-- our-shared-module
           |-- tracer.ts
    |-- some-other-module
           |-- node_modules
                |-- our-shared-module
                     |-- tracer.ts
```

The `some-other-module` was not using the tracer directly but because it was initialized automatically, I guess it was still executed.

To fix this problem we've updated the tracer file to export a function instead and we instantiate it when we want the tracer explicitly. I assumed the init should not be called more than once.
Let me know if that makes sense or not, I'm not sure if I'll get too much time to try to reproduce in a new project.

**Comment by bengl** [comment]
@vinnymac Were you able to reproduce locally with debug logs on?

**Comment by bengl** [comment]
@maximeloizeau your solution makes sense. Another approach might be to assign it to a global and check for presence there before initializing. Newer versions of the tracer also set a global for similar reasons.

**Comment by vinnymac** [comment]
@bengl Unfortunately I haven't had the time to look into this further. The system that experienced this issue is still reporting the problem on datadog APM. However, we are now running on v0.36.1

Would you be interested in setting up a call sometime to see if we could investigate this issue live together?

**Comment by rochdev** [comment]
We are currently in the process of trying a completely new route detection system. It's likely that this would fix your issue since it won't be prone to double routing as the current system can be.

@vinnymac We should have a release with that included soon if you want to try it.

**Comment by vinnymac** [comment]
@rochdev sounds great, as soon as it is available I’ll give it a try!

**Comment by rochdev** [comment]
@vinnymac We don't have an official release yet, but if you want to try the new routing detection logic you can install dd-trace `2.0.0-beta.0`. It's worth noting that it's a major release and includes a few minor breaking changes. While most applications shouldn't be affected, I would still recommend trying this in a non-prod environment first until we have the release notes out.

**Comment by rochdev** [comment]
We have now officially released 2.0 which should fix this issue. Please feel free to re-open if this is still happening with the latest version.
